File: GolangSolutions/Array/M1773_统计匹配检索规则的物品数量.go
package Array

var index = map[string]int{"type": 0, "color": 1, "name": 2}

func countMatches(items [][]string, ruleKey string, ruleValue string) (ans int) {
	idx := index[ruleKey]
	for _, item := range items {
		if ruleValue == item[idx] {
			ans++
		}
	}

	return
}

File: CPPSolutions/LeetCode/Array/M1773_统计匹配检索规则的物品数量.cpp
#include <bits\stdc++.h>
#include <unordered_map>

using namespace std;

class Solution {
public:
    int countMatches(vector<vector<string>>& items, string ruleKey, string ruleValue) {
        unordered_map<string, int> index = { {"type", 0}, {"color", 1}, {"name", 2} };
        int idx = index[ruleKey], ans = 0;
        for (auto&& item : items)
            if (ruleValue == item[idx])
                ans++;

        return ans;
    }
};